gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -0,0 +1,441 @@
+    mt = mt.annotate_rows(
+        qual_hists=hl.Struct(
+            **{
+                i.replace("_adj", ""): mt.qual_hists[i]
+                for i in mt.qual_hists
+                if "_adj" in i
+            }
+        ),
+        raw_qual_hists=hl.Struct(
+            **{i: mt.qual_hists[i] for i in mt.qual_hists if "_adj" not in i}
+        ),
+    )
+    return mt
+
+
+def generate_faf_popmax(mt: hl.MatrixTable) -> hl.MatrixTable:
+    """
+    Compute filtering allele frequencies and popmax with the AB-adjusted frequencies.
+
+    :param mt: Hail Table containing freq, ab_adjusted_freq, high_ab_het annotations.
+    :return: Hail MatirxTable with faf & popmax annotations.
+    """
+    faf, faf_meta = faf_expr(
+        mt.ab_adjusted_freq, mt.freq_meta, mt.locus, POPS_TO_REMOVE_FOR_POPMAX
+    )
+    mt = mt.annotate_rows(
+        faf=faf,
+        popmax=pop_max_expr(
+            mt.ab_adjusted_freq, mt.freq_meta, POPS_TO_REMOVE_FOR_POPMAX
+        ),
+    )
+    mt = mt.annotate_globals(
+        faf_meta=faf_meta,
+        faf_index_dict=make_faf_index_dict(faf_meta, label_delimiter="-"),
+    )
+    mt = mt.annotate_rows(
+        popmax=mt.popmax.annotate(
+            faf95=mt.faf[
+                mt.faf_meta.index(lambda x: x.values() == ["adj", mt.popmax.pop])
+            ].faf95
+        )
+    )
+    return mt
+
+
+def main(args):  # noqa: D103
+    # TODO: Determine if splitting subset freq from whole callset agg
+    subsets = args.subsets
+    test = args.test
+    chrom = args.chrom
+    af_threshold = args.af_threshold
+    adjust_freqs = args.adjust_freqs
+
+    hl.init(
+        log=f"/generate_frequency_data{'.' + '_'.join(subsets) if subsets else ''}.log",
+        default_reference="GRCh38",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+    # TODO: Update to release = True once outlier filtering is complete,
+    # possibly sample_meta=True if added
+    vds = get_gnomad_v4_vds(test=test)
+    meta_ht = meta.ht()
+    final_anns = []
+
+    logger.info("Adding metadata to VDS variant data cols...")
+    vds = hl.vds.VariantDataset(
+        vds.reference_data,
+        vds.variant_data.annotate_cols(meta=meta_ht[vds.variant_data.col_key]),
+    )
+
+    if test or chrom:
+        if test:
+            logger.info("Filtering to DRD2 in test VDS for testing purposes...")
+            test_interval = [
+                hl.parse_locus_interval("chr11:113409605-113475691")
+            ]  # NOTE: test on gene DRD2 for now, will update to chr22 when original PR goes in
+            vds = hl.vds.filter_intervals(vds, test_interval)
+            variants, samples = vds.variant_data.count()
+            logger.info(
+                "Test VDS has %s variants in DRD2 in %s samples...", variants, samples
+            )
+        else:
+            logger.info("Filtering to chromosome %s...")
+            vds = hl.vds.filter_chromosomes(vds, keep=f"chr{chrom}")
```

Comment:
Not that it matters for this run, but I prefer having the user pass in "chr20" rather than 20